internal/pkg/api/snow.go

```diff
@@ -117,3 +123,28 @@ func WithChangeForAllSnowMachines(change SnowMachineConfigFiller) SnowFiller {
 		}
 	}
 }
+
+// WithSnowIPPool sets a SnowIPPool.
+func WithSnowIPPool(name, ipStart, ipEnd, gateway, subnet string) SnowFiller {
```

Comment:
Do we already have unit test coverage here? If not, do you think it's worth adding?